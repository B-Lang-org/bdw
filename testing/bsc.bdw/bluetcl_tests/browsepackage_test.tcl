namespace import ::Bluetcl::*

# Loads the specified package
puts [bpackage load Test]

# Expands the information for the specified node
puts [browsepackage list 0]
puts [browsepackage list 1]
puts [browsepackage list 2]
puts [browsepackage list 3]
puts [browsepackage list 4]
puts [browsepackage list 10]
puts [browsepackage list 12]
puts [browsepackage list 11]

# Returns a detailed information for the specified node
puts [browsepackage detail 1]
puts [browsepackage detail 4]
puts [browsepackage detail 10]
puts [browsepackage detail 12]
puts [browsepackage detail 11]
puts [browsepackage detail 22]
puts [browsepackage detail 25]

# Returns the type of the specified node
puts [browsepackage nodekind 1]
puts [browsepackage nodekind 4]
puts [browsepackage nodekind 10]
puts [browsepackage nodekind 12]
puts [browsepackage nodekind 11]
puts [browsepackage nodekind 25]

# Loads the specified package
puts [bpackage load RegFile]

# Refreshes the package hierarchy
puts [browsepackage refresh]

# Shows the package hierarchy
puts [browsepackage list 0]

