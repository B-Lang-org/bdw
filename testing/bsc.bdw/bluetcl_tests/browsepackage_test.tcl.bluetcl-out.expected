Test Vector PrimArray List PreludeBSV Prelude
{1 . {}} {2 %/Libraries {}}
{3 Test {}}
{4 List {}} {5 Prelude {}} {6 PreludeBSV {}} {7 PrimArray {}} {8 Vector {}}

{9 {Functions and Modules} {}} {10 {Types and Interfaces} {}}
{11 f {}} {12 mkBVI {}} {13 mkM {}} {14 mkS {}} {15 mkT {}} {16 x {}}

{17 Bar {}} {18 Baz {}} {19 Foo {}} {20 SubIFC {}} {21 Sz {}} {22 T {}} {23 TopIFC {}} {24 U {}} {25 U2 {}}
.
Test {position {Test.bsv 1 1}}
Functions
function Test::f {result Bool} {arguments {Bool Bool}} {position {Test.bsv 55 15}}
Types
Alias Test::Sz {Definition 9} {position {Test.bsv 7 11}}
TaggedUnion {{Test::U#(type a)}} polymorphic {members {{void  F1  // width 0} {a  F2  } {List#(a)  F3  } {function Bool f(Bool x1)  F4  }}} {position {Test.bsv 35 3}}
Directory
Package
Function List
Function
Type List
Type
Test Vector RegFile PrimArray List PreludeBSV Prelude

{1 . {}} {2 %/Libraries {}}
